apps/andi/lib/andi/input_schemas/organizations.ex

```diff
@@ -30,6 +30,13 @@ defmodule Andi.InputSchemas.Organizations do
     Repo.all(query)
   end
 
+  def delete_harvested_dataset(dataset_id) do
+    case get_harvested_dataset(dataset_id) do
+      %{id: id} -> Repo.delete(%HarvestedDatasets{id: id})
+      _ -> Logger.info("Unable to delete dataset: #{dataset_id} from harvested datasets")
```

Comment:
You could also just check for `nil` and have the default case be a harvested dataset that you can delete instead of making a new one to delete